tsl/src/remote/async.c

```diff
@@ -561,38 +563,44 @@ get_single_response_nonblocking(AsyncRequestSet *set)
 		AsyncRequest *req = lfirst(lc);
 		PGconn *pg_conn = remote_connection_get_pg_conn(req->conn);
 
-		if (remote_connection_is_processing(req->conn) && req->state == DEFERRED)
-			return async_response_error_create("request already in process");
-
-		if (req->state == DEFERRED)
+		switch (req->state)
 		{
-			req = async_request_send_internal(req, WARNING);
+			case DEFERRED:
```

Comment:
Nit: This refactor seems to be changing semantics of Async API. The idea of `DEFERRED` state was that we can create a request that is not dispatched immediately. If I am not mistaken this refactor will send the request straight way so maybe we should rename this state to `CREATED` or similar.

Reply:
I don't think DEFERRED should be possible, so should be removed. But I didn't want to make further changes in this PR, instead focusing on fixing the subquery issues.

Let's discuss the use case for DEFERRED, if any, in another forum. But my guess is that this can be removed.